src/_zkapauthorizer/recover.py

```diff
@@ -104,6 +104,7 @@ class StatefulRecoverer:
     """
 
     _state: RecoveryState = RecoveryState(stage=RecoveryStages.inactive)
+    _listeners: Iterable[Callable[[RecoveryState], object]] = field(default=())
```

Comment:
I made listeners an initialization-time thing and removed the API for adding more of them later.  This was enough for all consumers of the interface and removes the possibility for weird listeners-changed-during-broadcast behavior.